### PR TITLE
Remove remaining Jackson usages, some optimizations in Monitor Exporter

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Removes the remaining usages of Jackson in `azure-monitor-opentelemetry-exporter` and adds some performance optimizations.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/MetricDataMapper.java [hunk 1]
```diff
@@ -273,10 +273,10 @@ private static boolean isSuccess(String metricName, Long statusCode, boolean cap
     }
 
     private static boolean isClient(String metricName) {
-       return metricName.contains(".client.");
+        return metricName.contains(".client.");
```
Comment (heyams) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/MetricDataMapper.java:276]: looks like we missed one space. would be nice to have `spotlessApply` to handle this automatically. I know this is in the roadmap. do you know when that will happen for `sdk/monitor/*`?

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/localstorage/LocalFileCache.java [hunk 2]
```diff
@@ -50,8 +50,8 @@ Queue<File> getPersistedFilesCache() {
     // this will avoid data loss in the case of app crashes and restarts.
     private static List<File> loadPersistedFiles(File folder) {
         return FileUtil.listTrnFiles(folder).stream()
+            .filter(file -> !isExpired(file, TimeUnit.DAYS.toSeconds(2))) // filter before sorting
```
Comment (heyams) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/localstorage/LocalFileCache.java:53]: definitely a good idea 👍 smaller set to sort.

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/localstorage/LocalFileWriter.java [hunk 3]
```diff
@@ -111,7 +111,9 @@ private static void write(File file, String connectionString, List<ByteBuffer> b
 
             FileChannel fileChannel = fileOut.getChannel();
             for (ByteBuffer byteBuffer : buffers) {
-                fileChannel.write(byteBuffer);
+                while (byteBuffer.hasRemaining()) { // possible for the ByteBuffer to not be fully written in one call
```
Comment (heyams) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/localstorage/LocalFileWriter.java:114]: 👍

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulsePingSender.java [hunk 4]
```diff
@@ -162,7 +152,7 @@ private String buildPingEntity(long timeInMillis) throws JsonProcessingException
             pingEnvelope.setVersion(sdkVersion);
         }
         pingEnvelope.setTimeStamp("/Date(" + timeInMillis + ")/");
-        return mapper.writeValueAsString(pingEnvelope);
+        return pingEnvelope.toJsonString().replace("/", "\\/");
```
Comment (heyams) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulsePingSender.java:155]: can you add this comment here 
`// By default jackson doesn't escape forward slashes (`/`), but the quick pulse backend requires
 // them to be escaped.`

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/model/QuickPulseDependencyDocument.java [hunk 5]
```diff
@@ -94,4 +82,55 @@ public String getOperationName() {
     public void setOperationName(String operationName) {
         this.operationName = operationName;
     }
+
+    @Override
+    public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
```
Comment (heyams) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/model/QuickPulseDependencyDocument.java:87]: in the relative near future, we will start using swagger to auto-generate quickpulse models.

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/model/QuickPulseDocument.java [hunk 6]
```diff
@@ -63,4 +57,62 @@ public Map<String, String> getProperties() {
...
     }
+
+    @Override
+    public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
+        return jsonWriter.writeStartObject()
+            .writeStringField("__type", type)
+            .writeStringField("DocumentType", documentType)
+            .writeStringField("Version", version)
+            .writeStringField("OperationId", operationId)
+            .writeMapField("Properties", properties, JsonWriter::writeString, true)
```
Comment (heyams) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/model/QuickPulseDocument.java:68]: you can delete true. 
this requires us to modify the auto-generated code. to fix that we fixed our agent not to put empty value in the map for our telemetry item exporter pipelines.

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/model/QuickPulseDocument.java [hunk 7]
```diff
@@ -63,4 +57,62 @@ public Map<String, String> getProperties() {
...
+            .writeEndObject();
+    }
+
+    JsonWriter toJsonShared(JsonWriter jsonWriter) throws IOException {
+        return jsonWriter.writeStringField("__type", type)
+            .writeStringField("DocumentType", documentType)
+            .writeStringField("Version", version)
+            .writeStringField("OperationId", operationId)
+            .writeMapField("Properties", properties, JsonWriter::writeString, true);
```
Comment (heyams) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/model/QuickPulseDocument.java:77]: please remove true, `skipNullValue` parameter

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulsePingSender.java (hunk 4 above)
Comment (heyams) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulsePingSender.java:155]: comment was from deleted `CustomCharacterEscapes`.

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/MetricDataMapper.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/MetricDataMapper.java:276]: That can be added any time when we feel comfortable with Spotless formatting the code in Monitor OpenTelemetry Exporter. I can file another PR after this for that if you want.

Comment (heyams) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/MetricDataMapper.java:276]: sooner the better. we use spotless everywhere from java agent to otel upstream. it's a handy tool for us.

**Review by heyams — APPROVED** [review]

**Pull request merged**